test/acceptance/js/helpers/DocstoreApp.js

```diff
@@ -15,6 +15,13 @@ const app = require('../../../../app')
 require('logger-sharelatex').logger.level('error')
 const settings = require('settings-sharelatex')
 
+// treat unhandled promise rejections as failures
+process.on('unhandledRejection', (e) => {
```

Comment:
in 10.17+ this can be done with https://nodejs.org/docs/latest-v10.x/api/cli.html#cli_unhandled_rejections_mode

Reply:
Oh, that's a nice feature. I wonder if it's worth adding to the buildscripts in that case?